# Assignment 5.1

- Name: Victor Hugo Germano 
- Date: 09/30/2024

For this assignment, you will refer to the textbook to solve the practice exercises. **Use Python to answer any coding problems (not R, even if indicated in your textbook).** Use Jupyter Notebook, Google Colab, or a similar software program to complete your assignment. Submit your answers as a **PDF or HTML** file. As a best practice, always label your axes and provide titles for any graphs generated on this assignment. Round all quantitative answers to 2 decimal places.

In [29]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import gridspec
import seaborn as sns
import scipy.stats as stats

sns.set()

## Problem 5.1.

Introducing notation for a parameter, state the following hypotheses in terms of the parameter
values and indicate whether it is a null hypothesis or an alternative hypothesis.

(a) The proportion of all adults in the UK who favor legalized gambling equals 0.50.


`Your answer goes here`

### Proportion of all adults

Null Hypothesis $H_0=0.5 $: 
- The proportion of all adults in the UK who favor legalized gambling equals 0.50.

Alternative Hypothesis $H_a\neq0.5$:
- The proportion of all adults in the UK who favor legalized gambling does not equal 0.50.


(b) The correlation for Australian adults between smoking (number of cigarettes per day) and blood pressure is positive.

`Your answer goes here`

### Correlation between smoking and blood pressure

Null Hypothesis $H_0: p = 0$: 
- There is no correlation between smoking and blood pressue

Alternative Hypothesis $H_a: p > 0$:
- There is a positive correlation between smoking and blood pressue


(c) The mean grade point average this year of all college graduates in the U.S. is the same for females and males.

`Your answer goes here`

### Correlation between smoking and blood pressure

Null Hypothesis $H_0: \mu_m = \mu_h$: 
- Males and Females have the same mean grade point average

Alternative Hypothesis $H_a: \mu_m \neq \mu_h$:
- Males and Females do not have the same mean grade point average


## Problem 5.6.

Before a Presidential election, polls are taken in two swing states. The Republican candidate
was preferred by 59 of the 100 people sampled in state A and by 525 of 1000 sampled in state
B. Treat these as independent binomial samples, where the parameter $\pi$ is the population
proportion voting Republican in the state.


(a) If we can treat these polls as if the samples were random, use significance tests of $H_0$:
$\pi  = 0.50$ against $H_a:  \pi > 0.50$ to determine which state has greater evidence supporting a
Republican victory. Explain your reasoning.


`Your answer goes here`

In [9]:
# state A
k1 = 59
n1 = 100
p_a = k1/n1

# State B
k2 = 525 
n2 = 1000
p_b = k2/n2

# mean of both samples
p_hat = (k1 + k2)/(n1 + n2)

# Null Hypothesis H0
# There's no difference between the two proportions (p_a = p_b)

# Alternative Hypothesis Ha
# There's a diferrence between the proportions

# z testing
z = (p_a - p_b) / np.sqrt(p_hat*(1-p_hat)*(1/n1 + 1/n2))

print("Z test result: %.3f"%z)

Z test result: 1.242


### Interpretation

Arriving at a Z value of 1.242 we reach at a p-value of 0.3925, using the Normal Distribution Table.

### Conclusion

Given a p-value of 0.3925, the conclusion can be drawn based on the comparison with $\pi=0.5$, we fail to reject the null hypothesis.

there is not enough statistical evidence to support a significant difference or effect as proposed by the alternative hypothesis, the data does not provide strong enough evidence to conclude which State has greater evidence insupport of Republican victory, even thought the higher proportion of support (0.59 vs. 0.525) indicates a stronger inclination towards Republican preferences in State A.

---


(b) Conduct a Bayesian analysis to answer the question in (a) by finding in each case the
posterior $P(\pi < 0.50)$, corresponding to the *P-* value in (a). Use beta(50, 50) priors, which
have standard deviation 0.05 and reflect the pollster’s strong prior belief that $\pi$ almost
surely is between 0.35 and 0.65. Explain any differences between conclusions.

`Your answer goes here`

### Answer
Bayesian Analysis:

Given a prior distribution: Beta (50, 50) which reflects the pollers belief that is between 0.35 and 0.65.

For both States(s) arriving at the Posterior distribution equation: 

$
 P(p_s|x_s) - Beta(x_s+alpha, n_s - x_s + beta) 
$

#### State A
- Prior:  Beta(50,50)
- Posterior: Beta(109, 91)
- $E[p_A] = \frac{109}{109+91} = 0.545$
- Bounds for state A: (0.4758, 0.6133)

#### State A
- Prior:  Beta(50,50)
- Posterior: Beta(575, 525)
- $E[p_B] = \frac{575}{575+525} = 0.523$
- Bounds for state B: (0.4932, 0.5522)

### Interpretation
 The posterior mean for State A (0.545) is higher than that for State B (0.523). This suggests that there is a greater proportion of support for the Republican candidate in State A compared to State B.
 For State A, the credible interval (0.4758, 0.6133) indicates that we are 95% confident that the true proportion of support lies within this range. For State B, the credible interval (0.4932, 0.5522) indicates a narrower range of support.
 Bayesian analysis indicates that State A has a higher estimated mean proportion of support compared to State B.

### Comparison with both methods
P-Value Interpretation: A failure to reject the null hypothesis does not imply that the null hypothesis is true, merely indicating that there is not enough evidence against it.

The Bayesian approach provides credible intervals, which directly convey uncertainty about parameter estimates, improving the analysis.


In [19]:
pi=0.5
# beta(50,50)
alpha, beta=50,50

# state A
xa, na = 59,100
p_a = xa/na

# State B
xb, nb = 525,1000 
p_b = xb/nb


# Applying beta distribution
alpha_a, beta_a = xa + alpha, na - xa + beta

alpha_b, beta_b = xb + alpha, nb - xb + beta


# Calculate quantiles for a 95% credible interval
lower_bound_a = stats.beta.ppf(0.025, alpha_a, beta_a)
upper_bound_a = stats.beta.ppf(0.975, alpha_a, beta_a)
print("Bounds for state A: %.4f,%.4f"%(lower_bound_a,upper_bound_a ))
# Calculate quantiles for a 95% credible interval
lower_bound_b = stats.beta.ppf(0.025, alpha_b, beta_b)
upper_bound_b = stats.beta.ppf(0.975, alpha_b, beta_b)
print("Bounds for state B: %.4f,%.4f"%(lower_bound_b,upper_bound_b ))

Bounds for state A: 0.4758,0.6133
Bounds for state B: 0.4932,0.5522


## Problem 5.8.

For the `Students` data file at the text website, analyze political ideology.

In [20]:
# Read in the Students data file
students = pd.read_csv('https://stat4ds.rwth-aachen.de/data/Students.dat', sep=r'\s+')

# View the first few rows of the dataset
students.head()

subject  gender  age  hsgpa  cogpa  dhome  dres    tv  sport  news  aids  \
0        1       0   32    2.2    3.5      0   5.0   3.0      5     0     0   
1        2       1   23    2.1    3.5   1200   0.3  15.0      7     5     6   
2        3       1   27    3.3    3.0   1300   1.5   0.0      4     3     0   
3        4       1   35    3.5    3.2   1500   8.0   5.0      5     6     3   
4        5       0   23    3.1    3.5   1600  10.0   6.0      6     3     0   

   veg  affil  ideol  relig  abor  affirm  life  
0    0      2      6      2     0       0     1  
1    1      1      2      1     1       1     3  
2    1      1      2      2     1       1     3  
3    0      3      4      1     1       1     2  
4    0      3      1      0     1       0     2

(a) Test whether the population mean $\mu$ differs from 4.0, the moderate response. Report the
*P*-value, and interpret. Make a conclusion using $\alpha$ - level = 0.05.

`Your answer goes here`

In [32]:
def interpret_p_value(p_value, alpha=0.05):
    if p_value < alpha:
        return "Reject the null hypothesis. There is a statistically significant difference."
    else:
        return "Fail to reject the null hypothesis. There is no statistically significant difference."

def calculate_p_value(t_score, df):
    p_value = 2 * (1 - stats.t.cdf(np.abs(t_score), df))
    return p_value

In [42]:
# variable ideol
# H0: population_mean == sample_mean
mu = 4.0

sample_mean = np.mean(students["ideol"])
sample_size = len(students.index)
std = np.std(students["ideol"])
df = sample_size - 1

test_stat = (sample_mean - mu)/(std / np.sqrt(sample_size))

p_value = calculate_p_value(test_stat, df)

result = interpret_p_value(p_value)

print("t stat: %.4f"%test_stat)
print("p-value: %.4f"% p_value)
print(result)

t stat: -4.6152
p-value: 0.0000
Reject the null hypothesis. There is a statistically significant difference.


(b) Construct the 95% confidence interval for $\mu$. Explain how results relate to those of the
test in (a).


`Your answer goes here`

In [41]:
# finding confindence interval 
sample_se = stats.sem(students['ideol'])
lower, upper = stats.norm.interval(confidence=0.95, loc=sample_mean, scale=sample_se)

print("Confidence Interval: (%.3f, %.3f)"%(lower, upper))

Confidence Interval: (2.619, 3.447)


## Problem 5.10.

A study of sheep mentioned in Exercise 1.27 analyzed whether the sheep survived for a year
from the original observation time (1 = yes, 0 = no) as a function of their weight (*kg*) at the
original observation. Stating any assumptions including the conceptual population of interest,
use a *t* test with the data in the Sheep data file at the text website to compare mean weights
of the sheep that survived and did not survive. Interpret the *P*-value.

`Your answer goes here`

In [51]:
data_url = 'https://stat4ds.rwth-aachen.de/data/Sheep.dat'
sheep_data = pd.read_csv(data_url, sep=r'\s+')
df = pd.DataFrame(sheep_data)

print(sheep_data.head())

   sheep  weight  survival
0      1    20.8         0
1      2    23.0         1
2      3    28.0         1
3      4    27.5         1
4      5    26.0         0


In [53]:
# Defining samples
survived_df= df[df['survival'] == 1]
died_df = df[df['survival'] == 0]

print("Survived sheep mean weight: %.3f"%survived_df['weight'].mean())
print("Lost sheep mean weight: %.3f"%died_df['weight'].mean())


t_stat, p_val = stats.ttest_ind(survived_df['weight'], died_df['weight'])
print("t value = %.2f, p val = %.2f"%(t_stat, p_val))

print(interpret_p_value(p_val))

Survived sheep mean weight: 20.646
Lost sheep mean weight: 15.998
t value = 14.50, p val = 0.00
Reject the null hypothesis. There is a statistically significant difference.


## Problem 5.11.

Use descriptive statistics and significance tests to compare the population mean political ideology for each pair of groups in Table 5.2 using the `Polid` data file. Summarize results using
*P*-values and using a non-technical explanation.

`Your answer goes here`

In [69]:
# Read and create data frame
data_url = 'https://stat4ds.rwth-aachen.de/data/Polid.dat'
sheep_data = pd.read_csv(data_url, sep=r'\s+')
df = pd.DataFrame(sheep_data)
print(df.head())
# Summaryzing the data by Race
df.groupby("race").describe()

       race  ideology
1  hispanic         1
2  hispanic         1
3  hispanic         1
4  hispanic         1
5  hispanic         1


ideology                                             
            count      mean       std  min  25%  50%  75%  max
race                                                          
black       371.0  3.816712  1.298165  1.0  3.0  4.0  4.0  7.0
hispanic    369.0  4.089431  1.339258  1.0  3.0  4.0  5.0  7.0
white      1835.0  4.170572  1.458553  1.0  3.0  4.0  5.0  7.0

### Interpretation

From the information presented here, it is possible to say that Race might play a role into someone choosing an ideology. We can expect that given two race groups from the sample, there is the possibility of their ideology differ. More information is needed though, given the fact that we cannot yet describe the effect size of this in real world scenarios.


In [121]:
def ttest_info(df1, df2):
    t_stat, p_val = stats.ttest_ind(ideol_by_race[0], ideol_by_race[1])
    print("Tests: p value: t = %.2f, p value = %.2f"%(t_stat, p_val))
    return t_stat, p_val

In [122]:


blk_df, his_df, whi_df = df[df['race'] == 'black'], df[df['race'] == 'hispanic'], df[df['race'] == 'white']


print("=== Hispanics and Blacks ===")
tval, pval = ttest_info(his_df, blk_df)
print(interpret_p_value(pval));print()

print("=== Hispanics and Whites ===")
tval, pval = ttest_info(his_df, whi_df)
print(interpret_p_value(pval));print()

print("=== Whites and Blacks ===")
tval, pval = ttest_info(whi_df, blk_df)
print(interpret_p_value(pval))

=== Hispanics and Blacks ===
Tests: p value: t = -2.81, p value = 0.01
Reject the null hypothesis. There is a statistically significant difference.

=== Hispanics and Whites ===
Tests: p value: t = -2.81, p value = 0.01
Reject the null hypothesis. There is a statistically significant difference.

=== Whites and Blacks ===
Tests: p value: t = -2.81, p value = 0.01
Reject the null hypothesis. There is a statistically significant difference.


## Problem 5.14 (a).

The `Income` data file at the book’s website shows annual incomes in thousands of dollars for
subjects in three racial-ethnic groups in the U.S.

(a) Stating all assumptions including the relative importance of each, show all steps of a significance test for comparing population mean incomes of Blacks and Hispanics. Interpret.


`Your answer goes here`

### Interpretation

Comparing the mean incomes of Blacks and Hispanics we need to execute a t test using both ethinicitis information.

With the result t = -2.81, p value = 0.01 we can assume that:
- Reject the Null Hypothesis:  statistically significant difference between income means 
- There is a significant effect in the data, meaning that ethnicity plays a role in income. 



In [123]:
data_url = 'https://stat4ds.rwth-aachen.de/data/Income.dat'
data = pd.read_csv(data_url, sep=r'\s+')
df = pd.DataFrame(data)

ethinicity = df.groupby("race")

# Descriptive Data Information
print(ethinicity.describe());print()

# Null Hypothesis H0: black_mean_income != black_mean_income
black_income = ethinicity.get_group("B")["income"]
hispanic_income = ethinicity.get_group("H")["income"]


ttest, pval = ttest_info(black_income, hispanic_income)
print(interpret_p_value(pval))


     income                                                  education  \
      count   mean        std   min   25%   50%   75%    max     count   
race                                                                     
B      16.0  27.75  13.284076  16.0  19.5  24.0  31.0   66.0      16.0   
H      14.0  31.00  12.812254  16.0  20.5  30.0  32.0   58.0      14.0   
W      50.0  42.48  22.869854  18.0  24.0  37.0  50.0  120.0      50.0   

                                                          
           mean       std  min    25%   50%    75%   max  
race                                                      
B     12.250000  3.356586  7.0  10.00  11.5  15.25  19.0  
H     11.642857  2.307418  8.0  10.00  11.0  12.00  16.0  
W     13.120000  2.804079  7.0  11.25  12.5  15.75  20.0  

Tests: p value: t = -2.81, p value = 0.01
Reject the null hypothesis. There is a statistically significant difference.


## Problem 5.15.

A recent report $\text{}^{39}$ estimated mean adult heights in the U.S. of 175.4 *cm* (69.1 inches) for men
and 161.7 *cm* (63.7 inches) for women, with standard deviation about 7 *cm* for each group. For
all finishers in the Boston Marathon since 1972, the time to finish has a mean of 221 minutes
for men and 248 minutes for women, each with a standard deviation of about 40 minutes.
According to the effect size, is the difference between men and women greater for height or for
marathon times? Explain.

`Your answer goes here`

### Interpretation

The difference in height between men and women has a large effect size, following Cohen's d interpretation, indicating a major difference at 1.9571.

The difference in marathon finish times has a medium effect size (Cohen's d of 0.675), indicating a moderate difference.

The large effect size for height sugests a greater gender difference than in marathon finishing times.

In [129]:
# effect size strengh of the phenomenon

def calculate_effect_size(mean1, mean2, std):
    return (mean1 - mean2) / std

man_mean = 175.4 #cm
women_mean = 161.7 #cm
height_std = 7

effect_size1 = calculate_effect_size(man_mean, women_mean, height_std)

man_finishers_mean = 221 #min
women_finishers_mean = 248 #min
marathon_std = 40

effect_size2 = calculate_effect_size(man_finishers_mean, women_finishers_mean, marathon_std)


(effect_size1, effect_size2)

(1.9571428571428595, -0.675)

## Problem 5.17.

Ideally, results of a statistical analysis should not depend greatly on a single observation. In a
sensitivity study, we re-do the analysis after deleting an outlier from the data set or changing its
value to a more typical value and checking whether results change much. For the anorexia data
analysis in Section 5.3.2, the weight change of 20.9 pounds for the cb group was a severe outlier.
Suppose this observation was actually 2.9 pounds but recorded incorrectly. Find the *P*-value
for testing $H_0: \mu1 = \mu2$ against $H_a: \mu1 \ne \mu2$ with and without that observation. Summarize its
influence.

`Your answer goes here`

## Problem 5.19.

In the 2018 General Social Survey, when asked whether they believed in life after death, 1017 of 1178 females said *yes*, and 703 of 945 males said *yes*. Test that the population proportions are equal for females and males. Report and interpret the *P*-value.

`Your answer goes here`

## Problem 5.23.

Use the `Happy` data file from the 2018 General Social Survey at the text website to form a
contingency table that cross classifies happiness with gender. For $ H_0 $: independence between
happiness and gender:

(a) Conduct and interpret the chi-squared test.

`Your answer goes here`

(b) Show the estimated expected frequencies and standardized residuals, and form a mosaic
plot. Explain how they are consistent with the result of the chi-squared test.

`Your answer goes here`

&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;


----------------
$^{39}$ See www.cdc.gov/nchs/data/nhsr/nhsr122-508.pdf and https://doi.org/10.1371/journal.pone.  